src/test/java/org/magicdgs/popgenlib/diversity/NucleotideDiversityUnitTest.java

```diff
@@ -0,0 +1,204 @@
+    }
+
+    @DataProvider(name = "piCounts")
+    public Object[][] getAlleleCountsForTajimasPi() {
+        // this is less exhaustive than the one with allele frequencies
+        // this is because it is exactly the same method, but computing the allele frequencies
+        return new Object[][] {
+                // monomorphic site with two samples
+                {Collections.singletonList(2), 0d},
+                {Arrays.asList(2, 0, 0), 0d},
+                // same counts
+                {Arrays.asList(5, 5), 0.5555555},
+                // tri-allelic
+                {Arrays.asList(5, 4, 1), 0.6444444}
+        };
+    }
+
+    @Test(dataProvider = "piCounts")
+    public void testTajimasPiWithCounts(final List<Integer> alleleCounts, final double expextedPi) {
+        Assert.assertEquals(NucleotideDiversity.tajimasPi(alleleCounts), expextedPi,
+                STATISTICAL_PRECISION);
+    }
+
+    @DataProvider(name = "piInvalidParamsWithFrequencies")
+    public Object[][] getInvalidParamsForTajimasPiWithFrequencies() {
+        final List<Double> alleleFreqs = Collections.singletonList(1d);
+        return new Object[][] {
+                // testing only invalid number of samples
+                {-1, alleleFreqs},
+                {0, alleleFreqs},
+                {1, alleleFreqs}
+        };
+    }
+
+    @Test(dataProvider = "piInvalidParamsWithFrequencies", expectedExceptions = IllegalArgumentException.class)
+    public void testInvalidTajimasPiWithFrequencies(final int numberOfSamples,
+            final List<Double> alleleFrequencies) throws Exception {
+        NucleotideDiversity.tajimasPi(numberOfSamples, alleleFrequencies);
+    }
+
+
+    @DataProvider(name = "thetaSegregatingSites")
+    public static Object[][] getSegregatingSitesForWattersonsTheta() {
+        // order of data:
+        // 1. number of samples
+        // 2. number of segregating sites
+        // 3. Expected Watterson's Theta
+        return new Object[][] {
+                // only 2 samples and 0 segregating sites (edge case)
+                {2, 0, 0d},
+                // only 2 samples and 1 segregating site (edge case)
+                {2, 1, 1d},
+                // for non-segregating sites is always 0 independently of the samples
+                {100, 0, 0d},
+                {10, 0, 0d},
+                // for only one segregating and different number of samples
+                {20, 1, 0.2818696},
+                {100, 1, 0.1931480},
+                // two segregating sites
+                {20, 2, 0.5637392},
+                {100, 2, 0.3862960},
+                // three segregating sites
+                {20, 3, 0.8456088},
+                {100, 3, 0.5794439},
+                // same number of segregating sites and samples
+                {20, 20, 5.6373922},
+                {100, 100, 19.3147978}
+                // TODO: test with very high sample size to test the approximation
```

Comment:
True! I added a test for sample size of 10,000 individuals and several number of segregating sites.